pkg/beacon/relay/event/events_tracking_test.go

```diff
@@ -0,0 +1,80 @@
+package event
+
+import (
+	"sync"
+	"testing"
+)
+
+func TestGroupSelectionTrack_Add(t *testing.T) {
```

Comment:
Instead of doing it this way and setting up `GroupSelectionTrack` state manually on the test setup, I'd rather prefer to implement those tests the same way they are implemented in `keep-ecdsa`, that is using a public API (`Add` + `Remove`)

Reply:
Sure, I can do the same way as in `keep-ecdsa`, but what are the advantages / disadvantages? We still use `GroupSelectionTrack.Add/Remove` in both cases. I thought that a series of related checks might be a good candidate for table tests. Anyway, open for both.

Reply:
Table tests are fine but I'd like to avoid setting internal state not using a public API and then making assumptions based on this state, e.g.:
```
groupSelected{Data: map[string]bool{"0x123456": true}, Mutex: &sync.Mutex{}},
```

What we want to test is that if `Add` is called twice with the same entry, it returns `true` for the first call and `false` for all subsequent calls until we `Remove`. Right now we are not testing this scenario. We are testing that if `{"0x123456": true}` is in `Data` we return `false` when called `Add` with `"0x123456"`. Technically, it's the same but practically, it's not what we wanted to test here. 

Also, if we structure the tests like in `keep-ecdsa` we treat `groupSelected` as a black box, we don't care what's happening inside as long as it's behaving as expected and we'll not have to modify unit tests if the internal implementation details change.

Reply:
done